### ISSUE TITLE
SQL Agent is unable to recognize the tables and its schema when connected to a Oracle DB.

### ISSUE THREAD

**Issue by madhubandru** [langchain-ai/langchain#16294]
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.

### Example Code

```
###Oracle DB Connection - Enable for Oracle DB
DIALECT = 'oracle'
SQL_DRIVER = 'oracledb'
oracle_conn_str=DIALECT + '+' + SQL_DRIVER + '://' + USER_ID + ':' + PASSWORD +'@' + HOST_NAME + ':' + str(PORT) + '/?service_name=' + SERVICE_NAME 
db_engine = create_engine(oracle_conn_str)
db = SQLDatabase(db_engine)


###Agent Code
sql_toolkit = SQLDatabaseToolkit(db=self.db, llm=llm)
sql_toolkit.get_tools()

self.agent = create_sql_agent(
  llm=llm,
  toolkit=sql_toolkit,
  agent_type=AgentType.ZERO_SHOT_REACT_DESCRIPTION,
  verbose=True,
  prefix=zero_prompt,
  extra_tools=[llm_tool],
  handle_parsing_errors=True,
  agent_executor_kwargs= {
    "return_intermediate_steps" : True
  }
)```

### Description

When using the SQL agent on Azure SQL Database, I provide the Database name in the connection string. This allows the SQL agent to successfully retrieve the table names and their schema.

However, when working with the SQL agent on Oracle Database, I provide the UserID, Password, Server name, Host, and Port in the connection string (provided code in example). The connection is established successfully, but the SQL agent encounters an issue in recognizing the tables. I suspect that the problem may arise from my tables being located under a specific schema (let's call it ABC_SCHEMA).

Upon investigation, it seems that when the SQL agent enters the executor chain with the "**Action: sql_db_list_tables**," it fails to list the tables under ABC_SCHEMA. As a result, an error is generated, specifically: **ValueError: An output parsing error occurred. In order to pass this error back to the agent and have it try again, pass handle_parse LLM output: Observation needed.**

In simple terms, the SQL agent is facing difficulty in identifying tables under a particular schema in Oracle Database, leading to the mentioned error. Can someone please help to fix this issue.

### System Info

langchain==0.0.348
oracledb==2.0.1
SQLAlchemy==2.0.25

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [X] Agents / Agent Executors
- [X] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

**Comment by madhubandru** [comment]
I tried to add schema in the connection string as suggested above but facing the below error.

sqlalchemy.exc.OperationalError: (oracledb.exceptions.OperationalError) DPY-6005: cannot connect to database (CONNECTION_ID=1MH+Ppcxxxx==).
DPY-6001: Service "SERVICE_NAME.ABC_SCHEMA" is not registered with the listener at host "HOST_NAME" port PORT_NUM. (Similar to ORA-12514)

(Background on this error at: https://sqlalche.me/e/20/e3q8)
        
 **Please Note:** Service name, host and port are masked.

**Comment by itsritiksingh** [comment]
Looking into your issue.

**Comment by wenshuili** [comment]
oracle_conn_str=DIALECT + '+' + SQL_DRIVER + '://' + USER_ID + ':' + PASSWORD +'@' + HOST_NAME + ':' + str(PORT) + '/?service_name=' + SERVICE_NAME + '.' + SCHEMA_NAME
is this work？schema_name can be specified in the link？

**Comment by wenshuili** [comment]
I have sloved the problem
connection_string = f"oracle+cx_oracle://{username}:{password}@{host}:{port}/{service_name}"
db = SQLDatabase.from_uri(connection_string, schema='wind')

**Comment by madhubandru** [comment]
@wenshuili Thank you for your response on this issue. I was using langchain's SQLDatabase(`from langchain_community.utilities import SQLDatabase`) function. This function has a parameter to pass the schema name, by which I was able to fix my issue.
